disperser/apiserver/server.go

```diff
@@ -605,6 +605,46 @@ func (s *DispersalServer) RetrieveBlob(ctx context.Context, req *pb.RetrieveBlob
 		return nil, api.NewInternalError("failed to get blob metadata, please retry")
 	}
 
+	// Check Ratelimit
```

Comment:
Move this before the L600, so it can protect invalid requests from hitting DynamoDB

Reply:
I moved the blob rate limit before the dynamo call. The throughput one comes after because we need the blob size.